### PR TITLE
Make tenant header configurable

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Related to https://github.com/jaegertracing/jaeger/pull/4151

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by pavolloffay** [comment]
Failed test `TestInstanceFailsLargeTracesEvenAfterFlushing`  should not be related to this PR.

File: cmd/tempo-query/tempo/config.go [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 import (
+	"github.com/jaegertracing/jaeger/plugin/storage/grpc/shared"
 	"github.com/spf13/viper"
 )
 
 // Config holds the configuration for redbull.
 type Config struct {
-	Backend string `yaml:"backend"`
+	Backend      string `yaml:"backend"`
+	TenantHeader string `yaml:"tenantHeader"`
```
Comment (mapno) [inline cmd/tempo-query/tempo/config.go:11]: In tempo we use snake_case for YAML names. Also, would it be better if it was named `tenant_header_key`?

**Review by mapno — COMMENTED** [review]
LGTM. Just left one comment on the case, and there are merge conflict. Once resolved we can merge.

File: cmd/tempo-query/tempo/config.go (hunk 1 above)
Comment (pavolloffay) [reply cmd/tempo-query/tempo/config.go]: fixed

**Review by mapno — APPROVED** [review]
LGTM, thanks!

**Pull request merged**